src/renderers/shared/reconciler/ReactChildReconciler.js

```diff
@@ -97,7 +99,8 @@ var ReactChildReconciler = {
         nextChildren[name] = prevChild;
       } else {
         if (prevChild) {
-          ReactReconciler.unmountComponent(prevChild, name);
+          removedChildren[name] = ReactReconciler.getNativeNode(prevChild);
```

Comment:
Do you mean that ReactChildReconciler would take some sort of callback to the renderer (or parent) and call it before unmounting? Would it do the same for mounting? I wasn't sure what temporal ordering you envisioned here.